### PR TITLE
Prohibit PADDING in response to ACK

### PR DESCRIPTION
**Pull request by martinduke**
Fixing #3100

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -3061,14 +3061,15 @@ of doing so.
...
 limits on how frequently they can be sent.  An endpoint MUST NOT send more than
-one ACK-frame-only packet in response to receiving an ack-eliciting packet (one
-containing frames other than ACK and/or PADDING).  An endpoint MUST NOT send a
-packet containing only an ACK frame in response to a non-ack-eliciting packet
-(one containing only ACK and/or PADDING frames), even if there are packet gaps
-which precede the received packet. Limiting ACK frames avoids an infinite
-feedback loop of acknowledgements, which could prevent the connection from ever
-becoming idle. However, the endpoint acknowledges non-ack-eliciting packets when
-it sends an ACK frame.
+one ACK-frame-only packet in response to receiving an ACK-eliciting packet
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: Maybe we should shorten this to "ACK-only packet" or even "ACK packet" and then define it at the beginning of the document?

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -3061,14 +3061,15 @@ of doing so.
...
+(one containing frames other than ACK and/or PADDING).  An endpoint MUST NOT
+send a packet containing only an ACK frame, or only an ACK and PADDING frame,
+in response to a non-ACK-eliciting packet (one containing only ACK and/or
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]: I'd remove the "one containing only ACK and/or PADDING frames)" now that it's well defined.

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -3061,14 +3061,15 @@ of doing so.
...
-one ACK-frame-only packet in response to receiving an ack-eliciting packet (one
-containing frames other than ACK and/or PADDING).  An endpoint MUST NOT send a
-packet containing only an ACK frame in response to a non-ack-eliciting packet
-(one containing only ACK and/or PADDING frames), even if there are packet gaps
-which precede the received packet. Limiting ACK frames avoids an infinite
-feedback loop of acknowledgements, which could prevent the connection from ever
-becoming idle. However, the endpoint acknowledges non-ack-eliciting packets when
-it sends an ACK frame.
+one ACK-frame-only packet in response to receiving an ACK-eliciting packet
+(one containing frames other than ACK and/or PADDING).  An endpoint MUST NOT
```
Comment (ianswett) [inline draft-ietf-quic-transport.md]:
```suggestion
.  An endpoint MUST NOT
```

File: draft-ietf-quic-transport.md (hunk 1 above)
Comment (martinduke) [reply draft-ietf-quic-transport.md]: Actually, non-ack-eliciting is already defined, and would make this flow much better.

File: draft-ietf-quic-transport.md [hunk 4]
```diff
@@ -3061,14 +3061,13 @@ of doing so.
...
-containing frames other than ACK and/or PADDING).  An endpoint MUST NOT send a
-packet containing only an ACK frame in response to a non-ack-eliciting packet
-(one containing only ACK and/or PADDING frames), even if there are packet gaps
-which precede the received packet. Limiting ACK frames avoids an infinite
-feedback loop of acknowledgements, which could prevent the connection from ever
-becoming idle. However, the endpoint acknowledges non-ack-eliciting packets when
-it sends an ACK frame.
+one ACK-frame-only packet in response to receiving an ack-eliciting packet.  An
+endpoint MUST NOT send a non-ack-eliciting packet in response to a non-ack-
+eliciting packet, even if there are packet gaps which precede the received
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]: A nit:  Though this hyphenation looks sane when presented this way, it won't be guaranteed to render nicely.  It might appear as `non-ack- eliciting` (note the extra space).  I think that you should remove the wrapping on the hyphen.
